scripts/release-checks/assert-valid-peer-dependency.mts

```diff
@@ -0,0 +1,54 @@
+import { Log, PackageJson, ReleasePrecheckError } from '@angular/ng-dev';
+import path from 'path';
+import url from 'url';
+import fs from 'fs';
+import semver from 'semver';
+
+interface PackageJson {
+  name: string;
+  peerDependencies: Record<string, string>;
+}
+
+/** Path to the current directory. */
+const currentDir = path.dirname(url.fileURLToPath(import.meta.url));
+
+const projectDir = path.join(currentDir, '../../');
+
+/**
+ * Ensures that the Angular framework peer dependency has been correctly updated
+ * to support given Angular versions. The following rules apply:
+ *
+ *    - TODO
+ */
+export async function assertValidFrameworkPeerDependency(newVersion: semver.SemVer) {
+  const major = newVersion.major;
+  const isPrerelease = !!newVersion.prerelease[0];
+  const isMajor = newVersion.minor === 0;
+
+  const ngDevkitBuildAngularPkgJson = await loadPackageJson(
+    path.join(projectDir, 'packages/angular_devkit/build_angular/package.json'),
+  );
+
+  let expectedFwDep: string;
+  if (isMajor) {
+    if (isPrerelease) {
+      expectedFwDep = `^${major}.0.0-next.0`;
+    } else {
+      expectedFwDep = `^${major}.0.0 || ^${major}.${newVersion.minor + 1}.0-next.0`;
+    }
+  } else {
+    expectedFwDep = `^${major}.0.0  || ^${newVersion.major}.${newVersion.minor + 1}.0-next.0`;
+  }
+
+  // TODO: check all angular packages.
+  // TODO: check `latest-versions.ts`.
```

Comment:
https://github.com/angular/angular-cli/blob/711214f210f10e4881b1bd419def9920e95a78ba/packages/schematics/angular/utility/latest-versions.ts contains the instruction we typically use and where they need to be updated.

I think we also need to check `ng-packagr` based on the last PR? https://github.com/angular/angular-cli/pull/24757

Reply:
I am not 100% certain about ng-packagr, just because currently it is a in different repo and releases are cut by myself. 

While I do cut an ng-packagr release prior to cutting the CLI. This might not always happen in the future. as I might be unavailable for the day either sick (Which technically I can still do a release if needed), be on vacation or have no internet access. At which point this check would cause the CLI release to be blocked.

Reply:
Discussed offline. We'll check `ng-packagr` but have an escape hatch.